### PR TITLE
feat(web): refactor color picker and add feature of remembering opaci…

### PR DESCRIPTION
**Pull request by majabnils**
…ty color when changing theme

# PR Checklist
https://elvia.atlassian.net/wiki/spaces/TEAMATOM/pages/10427498683/Review+prosess

- [ ] Bumpet package?
- [ ] Updated changelog?
- [ ] Correct date in changelog?

## Describe PR briefly:
https://elvia.atlassian.net/browse/LEGO-3499

### CODE REVIEW

File: packages/web/src/app/doc-pages/brand/color-doc/color-picker/color-picker.component.html [hunk 1]
```diff
@@ -1,20 +1,14 @@
...
   class="theme-container e-color-background-1"
-  [ngClass]="theme === 'dark' ? 'e-theme-dark' : 'e-theme-light'"
+  [ngClass]="theme() === 'dark' ? 'e-theme-dark' : 'e-theme-light'"
 >
   <div class="header-container">
-    <app-color-picker-header (changeThemeEvent)="handleChangeThemeEvent($event)" [currentTheme]="theme" />
+    <app-color-picker-header />
   </div>
-  <app-color-picker-exhibit [currentColor]="currentColor" [theme]="theme" [colorList]="colorList" />
+  <app-color-picker-exhibit [theme]="theme" />
```
Comment (TrygveUrdahl) [inline packages/web/src/app/doc-pages/brand/color-doc/color-picker/color-picker.component.html:8]: Is `theme` supposed to be here?
```suggestion
  <app-color-picker-exhibit />
```

File: packages/web/src/app/doc-pages/brand/color-doc/color-picker/color-picker-exhibit/color-picker-exhibit.component.ts [hunk 2]
```diff
@@ -30,40 +33,32 @@ import { ReplacePipe } from 'src/app/shared/pipes/replace.pipe';
...
-  }
-  private _currentColor: ColorElement;
-
-  get currentColorExistsInCurrentTheme(): boolean {
-    return !!getColorElement(this.currentColor.name, this.theme);
-  }
+    return !!getColorElement(this.currentColor()?.name, this.theme());
+  });
+  opacityColors = computed(() => {
+    if (this.currentColor === undefined) {
```
Comment (TrygveUrdahl) [inline packages/web/src/app/doc-pages/brand/color-doc/color-picker/color-picker-exhibit/color-picker-exhibit.component.ts:50]:
```suggestion
    if (this.currentColor() === undefined) {
```

File: packages/web/src/app/doc-pages/brand/color-doc/color-picker/color-picker-header/color-picker-header.component.ts [hunk 3]
```diff
@@ -12,5 +14,12 @@ import { LocalThemeSwitchComponent } from 'src/app/shared/local-theme-switch/loc
   schemas: [CUSTOM_ELEMENTS_SCHEMA],
 })
 export class ColorPickerHeaderComponent {
-  changeThemeEvent = output<ThemeName>();
+  private colorPickerService = inject(ColorPickerService);
+  theme = toSignal(this.colorPickerService.theme$, {
+    initialValue: 'light',
+  });
```
Comment (TrygveUrdahl) [inline packages/web/src/app/doc-pages/brand/color-doc/color-picker/color-picker-header/color-picker-header.component.ts:20]: Is this used anywhere?

Comment (majabnils) [reply packages/web/src/app/doc-pages/brand/color-doc/color-picker/color-picker-header/color-picker-header.component.ts]: I just use it to set the background of the main container

File: packages/web/src/app/doc-pages/brand/color-doc/color-picker/color-picker-color-list/color-picker-color-list.component.ts [hunk 4]
```diff
@@ -14,58 +16,38 @@ import { NeedsBorderPipe } from '../needs-border.pipe';
...
-  private _currentColor: ColorElement;
+  theme = this.colorPickerService.theme;
+  colorList = computed(() => (this.theme() === 'dark' ? darkColors : lightColors));
+  currentColor = this.colorPickerService.currentColor;
 
   chooseNewColor(color: ColorElement) {
-    this.chooseColor.emit(color);
-    this.userChoosesColor.emit();
+    this.colorPickerService.setCurrentColor(color.name);
   }
```
Comment (TomSchrier) [inline packages/web/src/app/doc-pages/brand/color-doc/color-picker/color-picker-color-list/color-picker-color-list.component.ts:29]: Nit: if this method only uses `color.name`, maybe we can adjust the parameter to `colorName`?

File: packages/web/src/app/doc-pages/brand/color-doc/color-picker/color-picker-exhibit/color-picker-exhibit.component.html [hunk 5]
```diff
@@ -35,20 +35,20 @@
...
       </app-copy>
 
       <!-- RGB -->
       <p class="e-title-caps">rgb</p>
       <p class="e-text-sm e-capitalize rgb">
-        @for (rgbValue of currentColor.rgb; track rgbValue; let isLast = $last) {
+        @for (rgbValue of currentColor()?.rgb; track rgbValue; let isLast = $last) {
           {{ rgbValue }}
           @if (!isLast) {
             <ng-container>/</ng-container>
```
Comment (TomSchrier) [inline packages/web/src/app/doc-pages/brand/color-doc/color-picker/color-picker-exhibit/color-picker-exhibit.component.html:54]: Do we need this `ng-container`?

**Review by TomSchrier — COMMENTED** [review]
LGTM 🧑‍🎨🎨

**Review by TomSchrier — APPROVED** [review]

**Pull request merged**